R/lint.R

```diff
@@ -58,7 +58,7 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
 
   if (isTRUE(parse_settings)) {
```

Comment:
I see, actually there's a non-trivial issue. We only run `read_settings()` here, but `get_lines()` above relies on some settings. Not immediately clear how to switch things around. If `encoding=` is the only issue, then we can just allow `encoding=NULL` to work in a few places. Reading more carefully...